OpenTreeMap/otm-core
# Refresh details when hitting "Save" on a plot/resource detail page

* Refactor templates and JS to allow reloading map feature details as a partial
* The partials (which extend `partials/map_feature_detail_base.html`) include the title area, action buttons, and detail area.
* The partials do not include the mini map and streetview, since that would cause an obvious flash
* Detail page tells inline edit form not to update on successful save (we'll wait for refresh)
* Show spinner on both "Save" and "Delete"
* Simplify code that was directly updating details

NOTE - looking at individual commits will make it easier to review the code, since there were a couple of re-indentation steps

Depends on OpenTreeMap/otm-addons#1230

Connects #2667
Connects #2664
Connects #1719
Connects #975

Testing
* Verify that the above open bugs are fixed
* Verify that these closed bugs are still fixed
  * #1585
  * #861
  * #627
  * #596
  * #396
* Verify that you can still add and delete a tree many times in succession
* Verify that after saving you can still edit and save everything
